### PR TITLE
Remove builtin clz in prvSelectHighestPriorityTask

### PR DESCRIPTION
**Pull request by chinglee-iot**
Remove builtin clz in prvSelectHighestPriorityTask

Description
-----------
__builtin_clz is compiler specific API should not be used in common code. This PR remove the usage of __builtin_clz.
We can have port optimization in another PR

Run RP2040 SMP demo with core affinity enabled.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: include/FreeRTOS.h [hunk 1]
```diff
@@ -352,6 +352,10 @@
     #define configNUM_CORES    1
 #endif
 
+#if ( configNUM_CORES > 32 )
```
Comment (aggarg) [inline include/FreeRTOS.h]: Is this true when `configUSE_CORE_AFFINITY` is 0? And should the check be `configUSE_CORE_AFFINITY > sizeof( UBaseType_t )`?

File: tasks.c [hunk 2]
```diff
@@ -1050,33 +1051,33 @@ static void prvAddNewTaskToReadyList( TCB_t * pxNewTCB ) PRIVILEGED_FUNCTION;
 
                     uxCoreMap &= ( ( 1 << configNUM_CORES ) - 1 );
 
-                    while( uxCoreMap != 0 )
+                    for( x = configNUM_CORES; x >= 0; x-- )
```
Comment (aggarg) [inline tasks.c]: Should the loop not be `for( x = 0; x < configNUM_CORES; x++ ) `? Seems like one extra iteration this way.

File: tasks.c [hunk 3]
```diff
@@ -1050,33 +1051,33 @@ static void prvAddNewTaskToReadyList( TCB_t * pxNewTCB ) PRIVILEGED_FUNCTION;
 
                     uxCoreMap &= ( ( 1 << configNUM_CORES ) - 1 );
 
-                    while( uxCoreMap != 0 )
+                    for( x = ( configNUM_CORES - 1 ); x >= 0; x-- )
```
Comment (chinglee-iot) [inline tasks.c:1054]: Indeed, the following is the possible value of UBaseType_t in FreeRTOS porting.
```
typedef portUBASE_TYPE UBaseType_t;
typedef uint32_t UBaseType_t;
typedef uint64_t UBaseType_t;
typedef uint8_t UBaseType_t;
typedef unsigned char UBaseType_t;
typedef unsigned long UBaseType_t;
typedef unsigned portBASE_TYPE UBaseType_t;
typedef unsigned short UBaseType_t;
```
We should use sizeof( UBaseType_t ) * CHAR_BIT to check. However, sizeof can't be used in preprocessor condition. The UBaseType_t size is decided by porting.
Suggest we remove the check here.

**Comment by chinglee-iot** [comment]
Indeed, the following is the possible value of UBaseType_t in FreeRTOS porting.
```
typedef portUBASE_TYPE UBaseType_t;
typedef uint32_t UBaseType_t;
typedef uint64_t UBaseType_t;
typedef uint8_t UBaseType_t;
typedef unsigned char UBaseType_t;
typedef unsigned long UBaseType_t;
typedef unsigned portBASE_TYPE UBaseType_t;
typedef unsigned short UBaseType_t;
```
We should use sizeof( UBaseType_t ) * CHAR_BIT to check. However, sizeof can't be used in preprocessor condition. The UBaseType_t size is decided by porting.
Suggest we remove the check here.

File: tasks.c (hunk 3 above)
Comment (chinglee-iot) [inline tasks.c:1054]: The original logic with builtin-clz favors higher core ID. So we decrease the iterator from ( coreNUM_CORS -1 ) to 0.

**Review by aggarg — APPROVED** [review]

**Pull request merged**